docs/website/contents/for-developers/HardWonWisdom.md

```diff
@@ -348,3 +348,77 @@ The key corresponding parts of the implementation are as follows.
+                - This constraint justifies that the function's type signature does not support applying it to a ticked ledger state; if it hypothetically did, then its output would exactly agree with the pre-image of that ticked ledger state.
+                  (Implementation detail: a ticked HFC ledger state must include `TransitionInfo` derived from the non-ticked ledger state only because the signature of `singleEraTransition` would not otherwise allow a ticked HFC ledger state's contents to determine the `TransitionInfo`.)
+                - Notably, this constraint is trivially true if the function is actually counting blocks in order to achieve its montonicity with respect to maximum rollbacks without relying on a Chain Growth property.
+
+See also "Which slots/times/epochs can we translate?" above.
+
+Some notable components are not listed above because they are determined by existing data about `F` and `G`.
+
+- The `TopLevelConfig (HardForkBlock [F, G])` can be completely determined from the same for `F` and `G`.
+- The initial state of the ledger and protocol can be completely determined from the same for `F`.
+
+### Eras that contain no slots
+
+The HFC interface and contract together permit `singleEraTransition cfg eraParams start st` to return `Just (boundEpoch start)`.
+In that case, the era contains no slots on the chain (and hence no blocks).
+It can only happen if `st` is the either the initial ledger state or else the result of translating the `LedgerState` from the latest block.
+The latest block isn't even necessarily in the preceding era, since that translation could have invoked multiple individual translation functions at once, if multiple eras contains no slots.
+
+It is not immediately obvious how this relates to the safe zone.
+Consider an era sequence `[F, G, H]`.
+If `G` has no slots, then the latest its end could have became known is once `F`'s end became known.
+Thus, the safe zone would seem to be respected as long as `G`'s safe zone is no larger than `F`'s.
+However, it may be too subtle that the block `x` in the definition of the safe zone is not necessarily from the current era!
+
+It's unclear if it's even desirable for the HFC to allow eras that contain no slots, with one exception: the Cardano system-level benchmarks already rely on this behavior in order to skip a prefix of the chain's eras.
+Indeed, skipping a prefix of eras seems like the most natural use case, since the "genesis block" itself is already not an actual block.
+But there is likely a more direct way to support that use case.
+And so it seems possible and desirable to simplify the HFC interface so that it only applies `singleEraTransition` to a ledger state arising from a block in the current era.
+Something alone the lines of "if an era's end is known before its first block exists, then the era must end before slot 0".
```

Comment:
:+1:, or phrased froma a different angle: the HFC has to support start from a ledger state that is not in the first era if all skipped eras contain zero slots.